app/models/katello/content_view_version.rb

```diff
@@ -240,12 +240,22 @@ def check_ready_to_promote!
     end
 
     def check_ready_to_destroy!
+      check_used_by_composite!
+
       if environments.any? && !organization.being_deleted?
         fail _("Cannot delete version while it is in environments: %s") % environments.map(&:name).join(",")
       end
       return true
     end
 
+    def check_used_by_composite!
+      if composite_content_views.any? && !organization.being_deleted?
+        fail _("Cannot delete version while it is in use by composite content views: %s") %
```

Comment:
It can be split in to multiple methods, if that is preferred.  This approach was taken simply to be consistent with similar methods in the class as well as it is currently only a couple of lines of logic.

As for doing it in destroy!, that would be nice; however, in this case, we cannot wait until the AR object is destroyed to perform the logic.  The reason being, backend objects (e.g. pulp) would have already been deleted by the time the validation is performed.  We have seen similar cases with other objects (e.g. repositories).